repos/spack_repo/builtin/packages/nextflow/package.py

```diff
@@ -11,137 +11,297 @@ class Nextflow(Package):
+            },
+        },
+        "22.10.4": {
+            "release": {
+                "sha256": "612a085e183546688e0733ebf342fb73865f560ad1315d999354048fbca5954d"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.10.3": {
+            "release": {
+                "sha256": "8d67046ca3b645fab2642d90848550a425c9905fd7dfc2b4753b8bcaccaa70dd"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.10.1": {
+            "release": {
+                "sha256": "fa6b6faa8b213860212da413e77141a56a5e128662d21ea6603aeb9717817c4c"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.10.0": {
+            "release": {
+                "sha256": "6acea8bd21f7f66b1363eef900cd696d9523d2b9edb53327940f093189c1535e"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.04.4": {
+            "release": {
+                "sha256": "e5ebf9942af4569db9199e8528016d9a52f73010ed476049774a76b201cd4b10"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.04.3": {
+            "release": {
+                "sha256": "a1a79c619200b9f2719e8467cd5b8fbcb427f43adf945233ba9e03cd2f2d814e"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.04.1": {
+            "release": {
+                "sha256": "89ef482a53d2866a3cee84b3576053278b53507bde62db4ad05b1fcd63a9368a"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+        "22.04.0": {
+            "release": {
+                "sha256": "8eba475aa395438ed222ff14df8fbe93928c14ffc68727a15b8308178edf9056"
+            },
+            "standalone": {
+                "sha256": "0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
+            },
+        },
+    }
 
-    depends_on("java", type="run")
+    for ver, attrs in nf_versions.items():
+        is_deprecated = False
+
+        if "deprecated" in attrs:
+            is_deprecated = True
+
+        # Edge release. Not preferred.
+        if "edge" in ver:
+            version(
+                ver,
+                sha256=attrs["release"]["sha256"],
+                expand=False,
+                deprecated=is_deprecated,
```

Comment:
Make the not preferred explicit. I don't think the default will ever change, but it is makes it clearer compared to the non-edge version.

```suggestion
                preferred=False,
                deprecated=is_deprecated,
```